tests/test_benchmarks.py

```diff
@@ -0,0 +1,151 @@
+EQX_ON_ERROR = "nan"  # TODO(jhaffner): will this do anything / work in test env?
+
+import functools as ft
+import warnings
+
+import jax
+import optimistix as optx
+import pytest
+import scipy as scp
+import sif2jax
+
+from .conftest import get_max_dimension
+
+
+scipy = pytest.mark.skipif("not config.getoption('scipy')")
+
+
+# Set a consistent number of maximum steps for all solvers. We might want to override
+# this, make it solver-specific in the future, or make it a config option.
+max_steps = 2**8
+
+# Benchmark solvers that are part of documented API.
+unconstrained_minimisers = (optx.BFGS(rtol=1e-3, atol=1e-6),)
+# Specify scipy minimisers using tuples (method_name: str, kwargs: dict)
+unconstrained_scipy_minimisers = (("BFGS", {}),)  # TODO(jhaffner): scipy tolerances?
+
+
+def get_test_cases(problems):
+    max_dimension = get_max_dimension()
+    if max_dimension is not None:
+        test_cases = []
+        for problem in problems:
+            problem_dimension = problem.y0().size
+            if problem_dimension <= max_dimension:
+                test_cases.append(problem)
+        return tuple(test_cases)
+    else:
+        return problems
+
+
+@pytest.mark.benchmark
+@pytest.mark.parametrize(
+    "problem", get_test_cases(sif2jax.unconstrained_minimisation_problems)
+)
+@pytest.mark.parametrize("minimiser", unconstrained_minimisers)
+def test_runtime_unconstrained_minimisers(benchmark, monkeypatch, minimiser, problem):
+    monkeypatch.setenv("EQX_ON_ERROR", "nan")
+    solve = jax.jit(
+        ft.partial(
+            optx.minimise,
+            problem.objective,
+            minimiser,
+            args=problem.args(),
+            max_steps=max_steps,
+            # We save on failure and filter during analysis to be able to compute the
+            # fraction of solved problems for performance profiles.
+            throw=False,
+        )
+    )
+
```

Comment:
```suggestion
y0 = jax.tree.map(lambda x: jax.device_put(x) if eqx.is_array(x) else x, problem.y0())
```
This makes sure that device transfer times are not includedi n benchmarks which can be significant for GPU's.